sql/plan/flush.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2022 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan
+
+import (
+	"fmt"
+
+	"github.com/dolthub/vitess/go/vt/sqlparser"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/grant_tables"
+)
+
+// Flush handles all flush statements. https://dev.mysql.com/doc/refman/8.0/en/flush.html
+type Flush struct {
+	option    *sqlparser.FlushOption
```

Comment:
It's not a hard rule, but generally we don't pass (or store) `sqlparser` values to our plan nodes. Instead we handle `sqlparser` in the `sql/parse` package. If need be, we'll create counterparts in either the `sql` or `sql/plan` packages that are a straight conversion from the `sqlparser` struct, but in other cases we can just pass the struct fields as parameters.